File: backend/models/Comment.ts
import { Schema, model, Types } from 'mongoose';

interface IComment {
  postId: Types.ObjectId;
  creator: string;
  content: string;
  createdAt: Date;
}

const commentSchema = new Schema<IComment>({
  postId: { type: Schema.Types.ObjectId, ref: 'Post', required: true },
  creator: { type: String, required: true, trim: true, minlength: 1 },
  content: { type: String, required: true, trim: true, minlength: 1 },
  createdAt: { type: Date, default: Date.now }
});

commentSchema.index({ postId: 1, createdAt: 1 });

const Comment = model<IComment>('Comment', commentSchema);

export { Comment };
export type { IComment };

File: mobile/lib/shared/models/comment.dart
class CommentModel {
  const CommentModel({
    required this.id,
    required this.postId,
    required this.creator,
    required this.content,
    required this.createdAt,
  });

  final String id;
  final String postId;
  final String creator;
  final String content;
  final DateTime createdAt;

  factory CommentModel.fromJson(Map<String, dynamic> json) {
    return CommentModel(
      id: json['_id'] as String? ?? '',
      postId: json['postId'] as String? ?? '',
      creator: json['creator'] as String? ?? '',
      content: json['content'] as String? ?? '',
      createdAt:
          DateTime.tryParse(json['createdAt'] as String? ?? '') ??
          DateTime.now(),
    );
  }
}
